src/arkode/arkode_relaxation.c

```diff
@@ -0,0 +1,844 @@
+/* -----------------------------------------------------------------------------
+ * Programmer(s): David J. Gardner @ LLNL
+ * -----------------------------------------------------------------------------
+ * SUNDIALS Copyright Start
+ * Copyright (c) 2002-2022, Lawrence Livermore National Security
+ * and Southern Methodist University.
+ * All rights reserved.
+ *
+ * See the top-level LICENSE and NOTICE files for details.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ * SUNDIALS Copyright End
+ * -----------------------------------------------------------------------------
+ * This is the implementation file for ARKODE's relaxation (in time)
+ * functionality
+ *
+ * Temporary vectors utilized in the functions below:
+ *   tempv1 - holds delta_y, the update direction vector
```

Comment:
Unfortunately, I think that use of `tempv1` here could destroy the routine `arkGetEstLocalErrors`, the copies the values of the local temporal error estimate from `tempv1` into `ele`.  Each of the ARKODE steppers fills `tempv1` with the local error estimate inside `*Step_ComputeSolutions`, under the assumption that nobody will modify this vector before `*Step_Evolve` returns control to the user.

That said, ARKODE has a `tempv4` that you don't seem to use here, and that should be available.

**Oops**: in looking around to write this response, I see now that `arkCheckConstrants` can also mess up `tempv1` in-between `*Step_ComputeSolutions` and when control is returned to the user.  We'll need to fix that as well (although I anticipate that either `tempv2` or `tempv3` could be used instead of `tempv1` in that routine).

Reply:
I've fixed this for relaxation and will open a PR shortly for `arkCheckConstaints`. It looks like `arkCheckConstraints` can mess up `GetEstLocalErrors` but only in a very specific case (when a step fails the constraint check that then immediately returns to the user because of hitting the max number of step attempt fails or min step size).

Reply:
Actually, I think that corner case is not possible because those returns happen before `tempv1` is used. So it would have to an even more unlikely scenario but regardless it should be fixed.